datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -298,13 +327,20 @@ pub(crate) async fn fetch_parquet_metadata(
         )));
     }
 
-    let footer_start = meta.size - 8;
+    let footer_start = if let Some(size_hint) = size_hint {
+        meta.size - size_hint
```

Comment:
What happens if size_hit is larger than the file size?

Maybe we could used `saturating_sub` here instead: https://doc.rust-lang.org/std/primitive.usize.html#method.saturating_sub